File: travel-planner-frontend/src/models/TravelPlan.js
export class TravelPlan {
  constructor(id, name, description, startDate, endDate, budget, notes, createdAt) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.startDate = startDate;
    this.endDate = endDate;
    this.budget = budget;
    this.notes = notes;
    this.createdAt = createdAt;
  }
}

File: TravelPlanner.ServiceFabric/TravelPlanner.TravelService/Models/TravelPlan.cs
using System.Diagnostics;

namespace TravelPlanner.TravelService.Models;

public class TravelPlan
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public decimal Budget { get; set; }
    public string Notes { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }

    public ICollection<Destination> Destinations { get; set; } = new List<Destination>();
    public ICollection<Activity> Activities { get; set; } = new List<Activity>();
    public ICollection<ChecklistItem> ChecklistItems { get; set; } = new List<ChecklistItem>();
}